lib/genesis_collector/megaraid.rb

```diff
@@ -0,0 +1,46 @@
+module GenesisCollector
+  module MegaRaid
+
+    def collect_megaraid
+      @payload[:raids] ||= []
+      @payload[:properties] ||= {}
+      unless read_megacli_adapter_type.nil?
+        @payload[:properties]['MEGARAID_TYPE'] = read_megacli_adapter_type
+        @payload[:properties]['MEGARAID_FW_PACKAGE'] = read_megacli_adapter_firmware
+        @payload[:raids].concat read_megacli_logical_disks
+      end
+    end
+
+    private
+
+    def read_megacli_adapter_type()
+      @megacli_adapter_output ||= shellout_with_timeout('megacli -AdpAllInfo -aAll', 10)
+      match = @megacli_adapter_output.match(/Product Name\s*:\s*(.+)\s*$/)
+      return nil if match.nil?
+      match[1]
+    end
+
+    def read_megacli_adapter_firmware()
+      @megacli_adapter_output ||= shellout_with_timeout('megacli -AdpAllInfo -aAll', 10)
+      match = @megacli_adapter_output.match(/FW Package Build\s*:\s*(\S+)$/)
+      return nil if match.nil?
+      match[1]
+    end
+
+    def read_megacli_logical_disks()
+      @megacli_ld_output ||= shellout_with_timeout('megacli -LDInfo -Lall -aAll', 10)
+      lds = []
+      @megacli_ld_output.split(/^$/).each do |disk|
+        lines = disk.split(/\n+/)
+        next if lines.count < 5 # skip status and empty
+        record = {}
+        lines .each do |line|
```

Comment:
whitespace between `lines` and `.each`. We usually have bots on our repos to catch this sort of thing. We should turn it on here sooner or later